### PR TITLE
Matter Pump: improve support for OperatingMode handling

### PR DESCRIPTION
**Pull request by hcarter-775**

### CODE REVIEW

File: drivers/SmartThings/matter-pump/profiles/pump-level.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ components:
     version: 1
   - id: switchLevel
     version: 1
+    config:
+      values:
+        - key: "level.value"
+          range: [1, 100]
```
Comment (HunsupJung) [inline drivers/SmartThings/matter-pump/profiles/pump-level.yml:12]: From Presentation team, I heard and checked when embedded config is applied, metadata cannot be applied.

File: drivers/SmartThings/matter-pump/src/init.lua [hunk 2]
```diff
@@ -175,9 +193,11 @@ local function on_off_attr_handler(driver, device, ib, response)
 end
 
 local function level_attr_handler(driver, device, ib, response)
-  if ib.data.value ~= nil then
-    local level = math.floor((ib.data.value / 254.0 * 100) + 0.5)
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.switchLevel.level(level))
+  if ib.data.value then
+      local max_level = get_field_for_endpoint(device, LEVEL_MAX, ib.endpoint_id) or DEFAULT_MAX_LEVEL
```
Comment (ctowns) [inline drivers/SmartThings/matter-pump/src/init.lua:197]: nit: unindent this by one tab (2 spaces)

File: drivers/SmartThings/matter-pump/src/init.lua [hunk 3]
```diff
@@ -175,9 +193,11 @@ local function on_off_attr_handler(driver, device, ib, response)
 end
 
 local function level_attr_handler(driver, device, ib, response)
-  if ib.data.value ~= nil then
-    local level = math.floor((ib.data.value / 254.0 * 100) + 0.5)
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.switchLevel.level(level))
+  if ib.data.value then
+      local max_level = get_field_for_endpoint(device, LEVEL_MAX, ib.endpoint_id) or DEFAULT_MAX_LEVEL
+      local level = math.floor((ib.data.value / max_level * MAX_CAP_SWITCH_LEVEL) + 0.5)
+      level = math.max(level, MIN_CAP_SWITCH_LEVEL)
```
Comment (ctowns) [inline drivers/SmartThings/matter-pump/src/init.lua:199]: This is the value we actually receive from the device, so I don't think we should change this value. If the device gives us 0, then that should be considered valid if that is how the device works

**Comment by hcarter-775** [comment]
Similar merge was made earlier today (6/26). Closing this PR.

**Pull request closed without merging**